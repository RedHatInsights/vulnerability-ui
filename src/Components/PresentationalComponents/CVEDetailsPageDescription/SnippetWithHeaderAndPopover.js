import { Popover, Stack, StackItem } from '@patternfly/react-core';
import propTypes from 'prop-types';
import React from 'react';
import Label from '../Snippets/Label';

const SnippetWithHeaderAndPopover = props => {
    const { title, value, content } = props;

    return (
        <Popover id="popover" bodyContent={content} headerContent={''} position="bottom"
            appendTo={document.querySelector('.vulnerability')}>
            <Stack className="popover-content">
                <StackItem>
                    <Label isLarge>{title}</Label>
                </StackItem>
                <StackItem>
                    <a>{value}</a>
                </StackItem>
            </Stack>
        </Popover>
    );
};

SnippetWithHeaderAndPopover.propTypes = {
    title: propTypes.any,
    value: propTypes.any,
    content: propTypes.any
};
export default SnippetWithHeaderAndPopover;
