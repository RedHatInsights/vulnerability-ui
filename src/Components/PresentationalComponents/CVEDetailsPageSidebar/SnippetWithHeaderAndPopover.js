import { Popover, Stack, StackItem } from '@patternfly/react-core';
import propTypes from 'prop-types';
import React from 'react';
import Label from '../Snippets/Label';

const SnippetWithHeaderAndPopover = ({ title, value, content }) => (
    <Popover id="popover" bodyContent={content} headerContent="" position="bottom"
        appendTo={document.querySelector('.vulnerability')}>
        <Stack className="popover-content">
            <StackItem>
                <Label isLarge>{title}</Label>
            </StackItem>
            <StackItem>
                <a>{value}</a>
            </StackItem>
        </Stack>
    </Popover>
);

SnippetWithHeaderAndPopover.propTypes = {
    title: propTypes.any,
    value: propTypes.any,
    content: propTypes.any
};

export default SnippetWithHeaderAndPopover;
