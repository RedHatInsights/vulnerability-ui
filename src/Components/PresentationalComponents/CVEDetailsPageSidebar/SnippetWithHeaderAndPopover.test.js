import React from 'react';
import SnippetWithHeaderAndPopover from './SnippetWithHeaderAndPopover';
import { render, screen } from '@testing-library/react';

const [title, value, tooltipText] = ['testTitle', 'testDescription', 'Some random text'];

describe('SnippetWithHeaderAndPopover', () => {
    it('Should render without params', () => {
        render(<SnippetWithHeaderAndPopover />);
        expect(screen.queryAllByText('testTitle')).toHaveLength(0);
        expect(screen.queryAllByAltText('testDescription')).toHaveLength(0);
        expect(screen.queryAllByAltText('Some random text')).toHaveLength(0);
    });
    it('Should render with title and label only', () => {
        render(<SnippetWithHeaderAndPopover title={title} value={value} />);
        expect(screen.getAllByText('testTitle')).toHaveLength(1);
        expect(screen.getAllByText('testDescription')).toHaveLength(1);
        expect(screen.queryAllByAltText('Some random text')).toHaveLength(0);
    });
    it.skip('Should render with title and label and Tooltip', () => {
        render(<SnippetWithHeaderAndPopover title={title} value={value} tooltipText={tooltipText} />);
        expect(screen.getAllByText('testTitle')).toHaveLength(1);
        expect(screen.getAllByText('testDescription')).toHaveLength(1);
        expect(screen.getAllByText('Some random text')).toHaveLength(1);
    });
});
