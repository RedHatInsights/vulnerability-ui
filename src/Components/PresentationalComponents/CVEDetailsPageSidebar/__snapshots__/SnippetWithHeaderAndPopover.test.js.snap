// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnippetWithHeaderAndPopover Should render with title and label and Tooltip 1`] = `
<SnippetWithHeaderAndPopover
  title="testTitle"
  tooltipText="Some random text"
  value="testDescription"
>
  <Popover
    appendTo={null}
    headerContent=""
    id="popover"
    position="bottom"
  >
    <Popper
      appendTo={null}
      distance={25}
      enableFlip={true}
      flipBehavior={
        [
          "top",
          "bottom",
          "left",
          "right",
          "top-start",
          "top-end",
          "bottom-start",
          "bottom-end",
          "left-start",
          "left-end",
          "right-start",
          "right-end",
        ]
      }
      isVisible={false}
      onDocumentClick={[Function]}
      onDocumentKeyDown={[Function]}
      onTriggerClick={[Function]}
      placement="bottom"
      popper={
        <ForwardRef
          active={false}
          aria-describedby="popover-popover-body"
          aria-label=""
          aria-modal="true"
          className="pf-c-popover"
          focusTrapOptions={
            {
              "clickOutsideDeactivates": true,
              "fallbackFocus": [Function],
              "returnFocusOnDeactivate": true,
              "tabbableOptions": {
                "displayCheck": "none",
              },
            }
          }
          onMouseDown={[Function]}
          preventScrollOnDeactivate={true}
          role="dialog"
          style={
            {
              "maxWidth": null,
              "minWidth": null,
              "opacity": 0,
              "transition": "opacity 300ms cubic-bezier(.54, 1.5, .38, 1.11)",
            }
          }
        >
          <PopoverArrow />
          <PopoverContent>
            <PopoverCloseButton
              aria-label="Close"
              onClose={[Function]}
            />
            
            <PopoverBody
              id="popover-popover-body"
            />
          </PopoverContent>
        </ForwardRef>
      }
      popperMatchesTriggerWidth={false}
      positionModifiers={
        {
          "bottom": "pf-m-bottom",
          "bottom-end": "pf-m-bottom-right",
          "bottom-start": "pf-m-bottom-left",
          "left": "pf-m-left",
          "left-end": "pf-m-left-bottom",
          "left-start": "pf-m-left-top",
          "right": "pf-m-right",
          "right-end": "pf-m-right-bottom",
          "right-start": "pf-m-right-top",
          "top": "pf-m-top",
          "top-end": "pf-m-top-right",
          "top-start": "pf-m-top-left",
        }
      }
      removeFindDomNode={false}
      trigger={
        <Stack
          className="popover-content"
        >
          <StackItem>
            <Label
              isLarge={true}
            >
              testTitle
            </Label>
          </StackItem>
          <StackItem>
            <a>
              testDescription
            </a>
          </StackItem>
        </Stack>
      }
      zIndex={9999}
    >
      <FindRefWrapper
        onFoundRef={[Function]}
      >
        <Stack
          className="popover-content"
        >
          <div
            className="pf-l-stack popover-content"
          >
            <StackItem>
              <div
                className="pf-l-stack__item"
              >
                <Label
                  isLarge={true}
                >
                  <span
                    className="vuln-label "
                    style={
                      {
                        "display": "block",
                        "fontSize": "medium",
                      }
                    }
                  >
                    testTitle
                  </span>
                </Label>
              </div>
            </StackItem>
            <StackItem>
              <div
                className="pf-l-stack__item"
              >
                <a>
                  testDescription
                </a>
              </div>
            </StackItem>
          </div>
        </Stack>
      </FindRefWrapper>
    </Popper>
  </Popover>
</SnippetWithHeaderAndPopover>
`;
